src/core/net/nat64_translator.hpp

```diff
@@ -76,6 +76,14 @@ class Translator : public InstanceLocator, private NonCopyable
         kDrop,          ///< The caller should drop the datagram silently.
     };
 
+    enum State : uint8_t
+    {
+        kStateDisabled = OT_NAT64_STATE_DISABLED, ///< The translator is disabled.
+        kStateNotRunning =
+            OT_NAT64_STATE_NOT_RUNNING, ///< The translator is not configured with a valid NAT64 prefix and a CIDR.
+        kStateActive = OT_NAT64_STATE_ACTIVE, ///< The translator is translating packets.
+    };
```

Comment:
We define similar `enum` here and also in `RoutingManager`?

We can define one common `enum` and re-use in both. 
Can define it in this header, before `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` and in `RoutingManager` header we already do  `#include "net/nat64_translator.hpp"`.